### ISSUE TITLE
@apply() does not seem to work within custom-style

### ISSUE THREAD

**Issue by mflorence99** [Polymer/polymer#2011]
In my element:

```css
  <style>
    :host {
      @apply(--mflo-spinning-numbers-theme); 
    }
  </style
```

In my document:

```css
  <style is="custom-style">
    :host {
      --mflo-spinning-numbers-theme {
        background-color: red;
        color: blue;
        display: inline-block;
        font-size: 32px;
        min-width: 280px;
      };
    }
  </style>
```

None of the styles are applied. Am I coding it incorrectly?

**Comment by herkulano** [comment]
`<style is="custom-style">` is a custom component, so it can't be applied in the main document like a style tag.

Have a look at [Polymer Starter Kit](https://developers.google.com/web/tools/polymer-starter-kit/index?hl=en) for an example implementation:
https://github.com/PolymerElements/polymer-starter-kit/blob/ba9ac0113c5a9f88bc5570e155beca91d122d5eb/app/elements/elements.html#L38 
https://github.com/PolymerElements/polymer-starter-kit/blob/master/app/styles/app-theme.html

More info here:
https://www.polymer-project.org/1.0/docs/devguide/styling.html#custom-style

**Comment by mflorence99** [comment]
Many thanks for your reply but I must be misunderstanding it as I am successfully using ```custom-style``` in the main document to style my element in exactly the way the docs you reference describe.

But I can only do that by creating specific custom properties in my element for each property I want to style, as in:

```css
:host {
  background-color: var(--mflo-spinning-numbers-background-color);
  color: var(--mflo-spinning-numbers-color);
  etc
}
```

As the Polymer docs point out, it's really impossible to enumerate all the properties so I want to use the mixin feature -- where I can define a grab bag and ```@apply()``` it as a whole.

I've been digging in the ```custom-style.html``` code and its dependencies but I haven't yet found out where ```@apply()``` might be handled. 

I'll keep digging but also take any help I can get :)

**Comment by sorvell** [comment]
The `:host` selector refers to the custom element itself and properties set there apply to the element's local dom. A `custom-style` has no local dom so it doesn't make sense to use `:host`. Instead if you'd like to set properties that apply to the entire document tree, you can use the `:root` selector. For example, try:

```
<style is="custom-style">
    :root {
      --mflo-spinning-numbers-theme {
        /* ... */
      };
    }
  </style>
```

**Comment by mflorence99** [comment]
Many thanks. Sadly, I still can't get ```@apply``` in my element to work using a *theme* set in the local DOM by ```custom-style``` -- but *custom properties* that my element uses via ```var()``` work just fine.

As the issue is closed I'll just assume it is me -- but I'll dig in the code more to try to see why.

**Comment by sorvell** [comment]
When you have more information, please feel free to re-open the issue or open a new one if necessary.

**Comment by justincy** [comment]
@mflorence99 I've run into the same problem. Did you ever figure out what was wrong?

**Comment by justincy** [comment]
Here's an example in Plunker: http://plnkr.co/edit/rmu87kemXD1D8jE6Rfs0?p=preview

What am I doing wrong?

**Comment by mflorence99** [comment]
Hey -- it has been a long time since I looked at that! But I must've, as my
code is working at: http://mflo-spinning-numbers.surge.sh/demo.html

So I looked back at what I did, and this works:

In my element:

  <style>
    :host {
      @apply(--mflo-spinning-numbers-theme);
    }
  </style>

In my document:

<style is="custom-style"> mflo-spinning-numbers {
--mflo-spinning-numbers-theme: { background-color: black; color: lightgreen;
display: inline-block; font-size: 32px; min-width: 280px; }; } </style>

If you compare that to what I wrote in the issue, I got the :host selector
wrong in the custom-style. Good luck!
